jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -642,20 +642,39 @@ static irep_idt get_if_cmp_operator(const irep_idt &stmt)
   throw "unhandled java comparison instruction";
 }
 
-static member_exprt to_member(const exprt &pointer, const exprt &fieldref)
+static member_exprt
+to_member(const exprt &pointer, const exprt &fieldref, const namespacet &ns)
 {
   struct_tag_typet class_type(fieldref.get(ID_class));
 
-  const typecast_exprt pointer2(pointer, java_reference_type(class_type));
+  const exprt typed_pointer =
+    typecast_exprt::conditional_cast(pointer, java_reference_type(class_type));
 
-  dereference_exprt obj_deref=checked_dereference(pointer2, class_type);
+  const irep_idt &component_name = fieldref.get(ID_component_name);
 
-  const member_exprt member_expr(
-    obj_deref,
-    fieldref.get(ID_component_name),
-    fieldref.type());
+  exprt accessed_object = checked_dereference(typed_pointer, class_type);
 
-  return member_expr;
+  // The field access is described as being against a particular type, but it
+  // may in fact belong to any ancestor type.
+  while(1)
+  {
+    struct_typet object_type =
```

Comment:
I'd use `const struct_typet &object_type = ...` here.

Reply:
This code is in fact simply moved - I'd prefer to follow this PR up with a clean up PR as this whole block I think can be replaced by `resolve_inherited_componentt`

Reply:
Sounds good!